src/user-guide/writing-workflows/scheduler.rst

```diff
@@ -0,0 +1,177 @@
+       [[events]]
+           mail events = stall, abort
+
+Email addresses and servers are configured by
+:cylc:conf:`global.cylc[scheduler][mail]`.
+
+Workflow event emails can be customised using
+:cylc:conf:`flow.cylc[scheduler][mail]footer`,
+:ref:`workflow_event_template_variables` can be used.
+
+For example to integrate with the Cylc 7 web interface (Cylc Review) the mail
+footer could be configured with a URL:
+
+.. code-block:: cylc
+
+   [scheduler]
+       [[events]]
+           mail footer = http://cylc-review/taskjobs/%(owner)s/?suite=%(workflow)s
+
+Custom Event Handlers
+^^^^^^^^^^^^^^^^^^^^^
+
+Cylc can also be configured to invoke scripts on workflow events.
+
+Event handler scripts can be stored in the workflow ``bin`` directory, or
+anywhere in ``$PATH`` in the :term:`scheduler` environment.
+
+They should return quickly to avoid tying up the scheduler process pool -
+see :ref:`Managing External Command Execution`.
+
+Contextual information can be passed to the event handler via
+:ref:`workflow_event_template_variables`.
+
+For example the following configuration will write some information to a file
+when a workflow is started:
+
+.. code-block:: bash
+   :caption: ~/cylc-run/<workflow-id>/bin/my-handler
+
+   #!/bin/bash
+
+   echo "Workflow $1 is running on $2:$3" > info
+
+.. code-block:: cylc
+   :caption: flow.cylc or global.cylc
+
+   [scheduler]
+       [[events]]
+           startup handlers = my-handler %(workflow)s %(host) %(port)
+
+.. _workflow_event_template_variables:
+
+Workflow Event Template Variables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. autoenumvalues:: cylc.flow.workflow_events.EventData
+
+
+.. _Managing External Command Execution:
+
+External Command Execution
+--------------------------
+
+Job submission commands, event handlers, and job poll and kill commands, are
+executed by the :term:`scheduler` in a "pool" of asynchronous
+subprocesses, in order to avoid blocking the workflow process.
+
+The process pool
+is actively managed to limit it to a configurable size, using
+:cylc:conf:`global.cylc[scheduler]process pool size`.
+
+Custom event handlers should be lightweight and quick-running because they
+will tie up a process pool member until they complete, and the workflow will
+appear to stall if the pool is saturated with long-running processes.
+However, to guard against rogue commands that hang indefinitely, processes
+are killed after a configurable timeout
+(:cylc:conf:`global.cylc[scheduler]process pool timeout`).
+
+All process kills are
+logged by the :term:`scheduler`. For killed job submissions, the associated
+tasks also go to the *submit-failed* state.
+
+
+Submitting Workflows To a Pool Of Hosts
+---------------------------------------
+
+By default ``cylc play`` will run workflows on the machine where the command
+was invoked.
+
+Cylc supports configuring a pool of hosts for workflows to run on,
+``cylc play`` will automatically pick a host and submit the workflow to it.
```

Comment:
I don't think we discuss anywhere the requirements for this pool.
The host you are submitting from and all hosts in the pool need to:
1. Share a common Cylc global config
2. Share a common HOME directory
3. Be set setup to allow passwordless ssh between all hosts

This may be better in the installation section?

Reply:
Not really anything to do with installation, may be better in a new section.

Reply:
Wasn't the goal of this PR but happy to add this here.

Reply:
I think of installation as what's required in order to setup Cylc ready for users to use (i.e. things admin users need to worry about). So, I think the installation section should at least gives pointers to info on run hosts and also platforms.
As you say, maybe not for this PR.

Reply:
Added your three points.

Reply:
Ah, Cylc installation as opposed to "user guide => installing workflow".

Reply:
I think we could do with an administrators section.

We could break it down as:

* For frontend users.
  * client commands
  * user interfaces
* For workflow developers.
  * user guide
  * workflow design guide
* For administrators.
  * advanced installation
  * conda recipes
  * system configuration
* For developers.
  * writing plugins

Reply:
Bumped to https://github.com/cylc/cylc-doc/issues/449